anoncreds/src/data_types/anoncreds/macros.rs

```diff
@@ -0,0 +1,62 @@
+#[macro_export]
+macro_rules! impl_anoncreds_object_identifier {
+    ($i:ident) => {
+        #[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize, Default)]
+        pub struct $i(pub String);
+
+        impl $i {
+            pub fn new(s: impl Into<String>) -> Self {
+                Self(s.into())
+            }
+
+            pub fn validated_new(
+                s: impl Into<String>,
+            ) -> Result<Self, crate::data_types::ValidationError> {
+                let s = Self(s.into());
+                s.validate()?;
+                Ok(s)
+            }
+        }
+
+        impl crate::data_types::Validatable for $i {
```

Comment:
It confuses me that we have this validation for id data types implemented, but all uses are removed, as far as I can tell?